src/components/SignIn.js

```diff
@@ -130,14 +130,27 @@ export const SignIn = hh(class SignIn extends Component {
       div({ style: { textAlign: 'center', height: '44px', width: '180px' } }, [
         img({ src: '/images/loading-indicator.svg', alt: 'spinner' })
       ]) :
-      h(GoogleLogin, {
-        scope: 'openid email profile',
-        theme: 'dark',
-        clientId: this.state.clientId,
-        onSuccess: this.responseGoogle,
-        onFailure: this.forbidden,
-        disabledStyle: { 'opacity': '25%', 'cursor': 'not-allowed' },
-      });
+      h(GoogleLogin,
+        _.isNil(this.props.customStyle) ? {
+          scope: 'openid email profile',
+          height: '44px',
+          width: '180px',
+          theme: 'dark',
+          clientId: this.state.clientId,
+          onSuccess: this.responseGoogle,
+          onFailure: this.forbidden,
+          disabledStyle: { 'opacity': '25%', 'cursor': 'not-allowed' }
+        } : {
+          render: (props) => button({className: 'btn-primary', onClick: props.onClick, style: this.props.customStyle}, 'Submit a Data Access Request'),
+          scope: 'openid email profile',
+          height: '44px',
+          width: '180px',
+          theme: 'dark',
+          clientId: this.state.clientId,
+          onSuccess: this.responseGoogle,
+          onFailure: this.forbidden,
+          disabledStyle: { 'opacity': '25%', 'cursor': 'not-allowed' }
+        });
```

Comment:
This is a good example where we can extract out the common set of properties between these two things and then merge in something additional for the changes. For example, scope, height, etc. are all the exact same values. To clean this up a bit, add this before the return statement on line 129:
```
    const defaultStyle = {
      scope: 'openid email profile',
      height: '44px',
      width: '180px',
      theme: 'dark',
      clientId: this.state.clientId,
      onSuccess: this.responseGoogle,
      onFailure: this.forbidden,
      disabledStyle: { 'opacity': '25%', 'cursor': 'not-allowed' }
    };
```
Then, you can replace the google login component like this:
```
      h(GoogleLogin,
        _.isNil(this.props.customStyle) ? defaultStyle : {
          ...{render: (props) => button({className: 'btn-primary', onClick: props.onClick, style: this.props.customStyle}, 'Submit a Data Access Request')},
          ...defaultStyle
        });
```
And the end result is that you have defined one set of component properties but used it in two places.

Reply:
https://www.javascripttutorial.net/object/javascript-merge-objects/